src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1636,6 +1636,11 @@ internal static CollectionExpressionTypeKind GetCollectionExpressionTypeKind(CSh
                     return CollectionExpressionTypeKind.Array;
                 }
             }
+            else if (isSpanOrListType(compilation, destination, WellKnownType.System_Collections_Immutable_ImmutableArray_T, out elementType)
+                && compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_InteropServices_ImmutableCollectionsMarshal__AsImmutableArray_T) is not null)
+            {
+                return CollectionExpressionTypeKind.ImmutableArray;
```

Comment:
Because we're not using `Binder.GetWellKnownTypeMember`, we're not registering any use-site diagnostics for this member. Please add a test for when `AsImmutableArray` has a use-site diagnostic of some kind.

Reply:
And also when `ImmutableCollectionsMarshal` has a use-site diagnostic, such as `CompilerFeatureRequired` applied.

Reply:
The other thing that will break here is used-assembly-references: please add a test for that where `ImmutableCollectionsMarshal` comes from a different assembly than `ImmutableArray` and verify that the assembly is correctly tracked.

Reply:
I'm fine with making the proposed adjustments, but would point out that other well-known member usages in this area, such as `CollectionsMarshal.SetCount`, are also not registering use-site diagnostics. Perhaps we should have a tracking item for fixing this across the board.